Flaskerizer_src/gui_for_flaskerizer.py

```diff
@@ -0,0 +1,185 @@
+        self.is_test = is_test
+        if not self.is_test:        #For testing purposes, make it unable to open the window.
+            self.root.mainloop()
+        else:
+            pass
+
+    def main_layout(self):
+        """The GUI layout, defines all the buttons,
+        labels, and entries seen on the window.
+        """
+
+        self.main = tk.Frame(self.root)
+        self.label_html = tk.Label(self.main,
+                                   text="HTML File Location:")
+        self.label_html.grid(row=0, column=0, sticky="WENS", pady=20)
+        self.folder_search_html = tk.Button(self.main,
+                                            text="Search Files", command=self.get_html_folder)
+        self.folder_search_html.grid(row=0, column=1, sticky="WENS", pady=20)
+        self.folder_location_html = tk.Entry(self.main,
+                                             textvariable=self.html_location, width=80)
+        self.folder_location_html.grid(row=0, column=2, sticky="WENS", pady=20)
+
+
+        self.label_static = tk.Label(self.main,
+                                     text="Static Folder Location:")
+        self.label_static.grid(row=1, column=0, sticky="WENS", pady=20)
+        self.folder_search_static = tk.Button(self.main,
+                                              text="Search Files", command=self.get_static_folder)
+        self.folder_search_static.grid(row=1, column=1, sticky="WENS", pady=20)
+        self.folder_location_static = tk.Entry(self.main,
+                                               textvariable=self.static_location, width=80)
+        self.folder_location_static.grid(row=1, column=2, sticky="WENS", pady=20)
+
+
+        self.label_js = tk.Label(self.main,
+                                 text="JavaScript File Location:")
+        self.label_js.grid(row=2, column=0, sticky="WENS", pady=20)
+        self.folder_search_js = tk.Button(self.main,
+                                          text="Search Files", command=self.get_js_folder)
+        self.folder_search_js.grid(row=2, column=1, sticky="WENS", pady=20)
+        self.folder_location_js = tk.Entry(self.main,
+                                           textvariable=self.js_location, width=80)
+        self.folder_location_js.grid(row=2, column=2, sticky="WENS", pady=20)
+
+
+        self.main.pack()
+        self.ok_button = tk.Button(self.root,
+                                   text="OK", command=self.get_values).pack()
+
+    def get_html_folder(self):
+        """Opens a file dialog prompting the user to select an HTML file.
+        Gets the directory path with the path_to_folder function
+        and sets the html_location variable with the returned path value.
+        """
+
+        path = filedialog.askopenfilename(title= "Select the main html file normally index.html",
+        filetypes=(("Html Files", "*.html"), ("all files", "*.*")))
+        self.html_location.set(path)
+        if self.validate_path(path):
```

Comment:
Here it seems to take the path in any case where it is a path to a real file. You could use the endswith('.html') string method to validate that they actually select and html file or something like that. Anytime you have user input be it a gui or a webform the law is "never trust the user". Someone will switch the accept all files, pick a .jpg, and then wonder why it is not working and we need to have things laid out in a way that prevents headaches. 

In my opinion it should be like this:

1. User opens dialog box and selects path to a valid .html file, the path gets presented in the box

2. User opens dialog box then backs out again right away without selecting anything "Select one HTML file from the template" gets recycled in the box, because nothing has changed really

3. User opens dialog box and picks anything other than and html file, some informative warning gets presented that they need to pick an html file